tutorials/cloud-run-with-pomerium-for-end-user-access/index.md

```diff
@@ -0,0 +1,175 @@
+
+- Add an IAM policy delegating `roles/run.invoker` permissions to a service account
+- Run Pomerium with access to a key for the corresponding service account
+- Publish DNS records for each protected application pointing to Pomerium
+- Configure Pomerium with appropriate policy and `enable_google_cloud_serverless_authentication`
+
+The protected application delegates trust to a GCP service account which Pomerium runs as, and Pomerium performs user based authorization on a per route basis. This turns Pomerium into a bridge between a user-centric and service-centric authorization models.
+
+## Before you begin
+
+This guide assumes you have Editor access to a Google Cloud project which can be used for isolated testing, and a DNS zone which you are also able to control. DNS does not need to be inside Google Cloud for the example to work.
+
+## Set Up
+
+To deploy Pomerium to Cloud Run, a special [image](https://console.cloud.google.com/gcr/images/pomerium-io/GLOBAL/pomerium) is available at `gcr.io/pomerium-io/pomerium-[version]-cloudrun`. It allows sourcing configuration from GCP Secrets Manager, and sets some defaults for Cloud Run to keep configuration minimal. We will be leveraging it in this example to store identity provider credentials. Pomerium's [authorization policy](https://www.pomerium.com/reference/#policy) contains no secrets so we can place it directly in an [ENV var](https://www.pomerium.io/reference/#configuration-settings).
+
+[Dockerfile](https://github.com/pomerium/pomerium/blob/master/.github/Dockerfile-cloudrun) Based on [vals-entrypoint](https://github.com/pomerium/vals-entrypoint)
+
+The image expects a config file at `/pomerium/config.yaml`. Set `VALS_FILES=[secretref]:/pomerium/config.yaml` and any other [Pomerium Environment Variables](https://www.pomerium.io/reference/#configuration-settings) directly or with secret refs such as `ref+gcpsecrets://PROJECT/SECRET(#/key])`.
+
+## Config
+
+Set up Pomerium's [config.yaml](https://www.pomerium.com/reference/#shared-settings) to contain your identity provider credentials and secrets (config.yaml):
+
+```yaml
+# config.yaml
+authenticate_service_url: https://authn.cloudrun.pomerium.com
+shared_secret: XXXXXX
+cookie_secret: XXXXXX
+idp_provider: "google"
+idp_client_id: XXXXXX
+idp_client_secret: "XXXXXX"
+```
+
+Substitute `cloudrun.pomerium.com` for your own subdomain and your e-mail domain if appropriate (policy.template.yaml):
+
+```yaml
+# policy.template.yaml
+# see https://www.pomerium.com/reference/#policy
+- from: https://hello.cloudrun.pomerium.com
+  to: ${HELLO_URL}
+  allowed_domains:
+    - gmail.com
+  enable_google_cloud_serverless_authentication: true
+- from: https://httpbin.cloudrun.pomerium.com
```

Comment:
would be good to describe above in the doc somewhere that

> During this tutorial, we'll deploy a Pomerium proxy that allows @gmail.com users to authenticate and proxies traffic to httpbin.org.

otherwise the reader won't know what they would be reading.

Reply:
Done up top.